### PR TITLE
build: Use Go 1.19 for Terraform v1.3

### PR DESCRIPTION
**Pull request by apparentlymart**
This will cause our official releases for the forthcoming v1.3 series to be built with Go 1.19 instead of Go 1.18 as before. This is in line with our usual policy of adopting the latest Go minor release available at the time when we freeze each Terraform minor release.

---

Go 1.19's "fmt" has some awareness of the new doc comment formatting conventions and adjusts the presentation of the source comments to make it clearer how godoc would interpret them. Therefore this commit includes various updates made by "go fmt" to acheve that.

In line with our usual convention that we make stylistic/grammar/spelling tweaks typically only when we're "in the area" changing something else anyway, I also took this opportunity to review most of the comments that this updated to see if there were any other opportunities to improve them.

---

I intend to include the following in our own changelog to pass on a relevant caveat from the Go 1.19 Release Notes:

* When making requests to HTTPS servers, Terraform will now reject invalid handshakes that have duplicate extensions, as required by RFC 5246 section 7.4.1.4 and RFC 8446 section 4.2. This may cause new errors when interacting with existing buggy or misconfigured TLS servers, but should not affect correct servers.

    This only applies to requests made directly by Terraform CLI, such as provider installation and remote state storage. Terraform providers are separate programs which decide their own policy for handling of TLS handshakes.
    
There is also a change that _might_ affect Terraform's visible behavior, but I think not in a way that's significant enough to call out explicitly:

* The pure Go DNS resolver now uses a DNS extension to allow larger reply packets. The Go team acknowledges that this might interact poorly with old/buggy DNS servers, but doesn't seem to be super concerned about it.

### CODE REVIEW

**Comment by apparentlymart** [comment]
The Vercel build error here seems to be unrelated to this change and is complaining about an `ENOMEM` error when reading some files unrelated to this Go upgrade.

**Comment by apparentlymart** [comment]
Apparently something we're running in our code consistency checks is blocking us from using the deprecated functions in `io/ioutil`.

As discussed over in #31357, we were not planning to do a mass-update of those all at once in order to avoid suddently creating merge conflict hell for ourselves in case we need to backport to older versions. It seems that in order to proceed here we'll need to either change that decision or change the configuration of this checking tool to permit us to continue using these deprecated functions.

I would personally lean towards the latter, especially if we can combine it with some settings in our own editors to mark the deprecations as warnings so we can easily identify opportunities to improve when we're already in the area changing some nearby code anyway. But I'm not sure how practical either of those is, so will need to investigate further.

**Comment by apparentlymart** [comment]
I think retaining our existing policy ought to be the "default" answer here and so I've pushed a new commit to change the `staticcheck` options to agree with that policy.

We could still change the policy, but if we do that then it'll be in a separate PR which can both re-enable this rule in `staticcheck` _and_ bulk-update all of the existing deprecation errors to allow us to move forward.

File: internal/plans/internal/planproto/planfile.pb.go [hunk 1]
```diff
@@ -517,14 +517,14 @@ type Change struct {
...
 	// msgpack-encoded HCL values involved in the change.
-	// - For update and replace, two values are provided that give the old and new values,
-	//   respectively.
-	// - For create, one value is provided that gives the new value to be created
-	// - For delete, one value is provided that describes the value being deleted
-	// - For read, two values are provided that give the prior value for this object
-	//   (or null, if no prior value exists) and the value that was or will be read,
-	//   respectively.
-	// - For no-op, one value is provided that is left unmodified by this non-change.
+	//   - For update and replace, two values are provided that give the old and new values,
```
Comment (alisdair) [inline internal/plans/internal/planproto/planfile.pb.go:520]: These changes to the protobuf generated code are reverted when you run `make protobuf`. I guess we'll need to either update the proto files to match gofmt…?

I'm also surprised we don't have a check tracking that `make protobuf` results in no changes.

**Review by alisdair — COMMENTED** [review]
I think we should address the protobuf issues (see inline note), but otherwise this looks good to me.

File: internal/plans/internal/planproto/planfile.pb.go (hunk 1 above)
Comment (apparentlymart) [reply internal/plans/internal/planproto/planfile.pb.go:520]: Along the way here I initially _didn't_ run `make protobuf` and got an error from our "quick checks" workflow, so what you see here is the result of me running `make protobuf` after everything's upgraded to Go 1.19 versions in order to make tat pass.

Can you share what you tried that led you to seeing an error? I'm trying to understand if there's just something in your environment that isn't upgraded yet or if there's a more annoying inconsistency to deal with here. :thinking:

Comment (apparentlymart) [reply internal/plans/internal/planproto/planfile.pb.go:520]: The "Quick Checks" workflow is running everything all in a single `make` command like this:

```bash
make fmtcheck importscheck generate staticcheck exhaustive protobuf
```

I wonder if the problem here is that `make fmtcheck` and `make protobuf` are "fighting" one another by both expecting different results, and the check thing happens to be running them in the right order for my code here to pass. :thinking:

Comment (alisdair) [reply internal/plans/internal/planproto/planfile.pb.go:520]: Oh, oops, my mistake! I wasn't using go 1.19 in the terminal tab where I ran `make protobuf`, and confirm that it executes cleanly when I do.

**Review by alisdair — APPROVED** [review]

**Pull request merged**